x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/CloseJobRequest.java

```diff
@@ -163,29 +170,75 @@ public boolean equals(Object other) {
         }
 
         CloseJobRequest that = (CloseJobRequest) other;
-        return Objects.equals(jobIds, that.jobIds) &&
-            Objects.equals(timeout, that.timeout) &&
-            Objects.equals(allowNoJobs, that.allowNoJobs) &&
-            Objects.equals(force, that.force);
+        return Objects.equals(jobIds, that.jobIds) && Objects.equals(requestBody, that.requestBody);
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
-
         builder.field(JOB_IDS.getPreferredName(), jobIds);
+        requestBody.doXContent(builder);
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this);
+    }
 
-        if (timeout != null) {
-            builder.field(TIMEOUT.getPreferredName(), timeout.getStringRep());
+    public static final class RequestBody implements ToXContentObject {
```

Comment:
I don't think we need an inner class here. We can keep the optional params as they were. They are direct parts of the request object. Note that even the `job_ids` can be passed in the request body in theory.